# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/credentialmanagement/identity/PWebIdentity.ipdl
# Commit: 46af55564501
# Full Hash: 46af555645013a126416a440b658f2969eca4b20
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-06-06 08:52:03
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251617
# ==============================================================================

diff -r 55c4afa3a46c -r 46af55564501 dom/credentialmanagement/identity/PWebIdentity.ipdl
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/credentialmanagement/identity/PWebIdentity.ipdl	Thu Jun 05 14:53:36 2025 +0000
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*
+ * IPC Transaction protocol for the FedCM DOM API.
+ */
+
+include protocol PWindowGlobal;
+
+include "mozilla/dom/IdentityCredentialSerializationHelpers.h";
+
+using mozilla::dom::CredentialMediationRequirement from "mozilla/dom/CredentialManagementBinding.h";
+using mozilla::dom::IdentityCredentialDisconnectOptions from "mozilla/dom/IdentityCredentialBinding.h";
+using mozilla::dom::IdentityCredentialRequestOptions from "mozilla/dom/IdentityCredentialBinding.h";
+using mozilla::dom::IdentityProviderConfig from "mozilla/dom/IdentityCredentialBinding.h";
+using mozilla::dom::LoginStatus from "mozilla/dom/LoginStatusBinding.h";
+
+
+namespace mozilla {
+namespace dom {
+
+struct IPCIdentityCredential
+{
+  nsString id;
+  nsString? token;
+};
+
+
+union WebIdentityGetCredentialResponse {
+  nsresult;
+  IPCIdentityCredential;
+};
+
+async protocol PWebIdentity {
+  manager PWindowGlobal;
+
+  parent:
+    async RequestCancel();
+
+  async GetIdentityCredential(IdentityCredentialRequestOptions aOptions,
+                              CredentialMediationRequirement aMediationRequirement,
+                              bool aHasUserActivation)
+    returns (WebIdentityGetCredentialResponse response);
+
+  async DisconnectIdentityCredential(IdentityCredentialDisconnectOptions aOptions) returns (nsresult rv);
+
+  async PreventSilentAccess() returns (nsresult rv);
+
+  async SetLoginStatus(LoginStatus foo) returns (nsresult rv);
+
+  child:
+    async __delete__();
+};
+
+}
+}